components/camel-paho-mqtt5/src/main/java/org/apache/camel/component/paho/mqtt5/PahoMqtt5Consumer.java

```diff
@@ -155,6 +160,23 @@ public Exchange createExchange(MqttMessage mqttMessage, String topic) {
         paho.setHeader(PahoMqtt5Constants.CAMEL_PAHO_MSG_PROPERTIES, mqttMessage.getProperties());
 
         exchange.setIn(paho);
+        if (getEndpoint().getConfiguration().isManualAcksEnabled()) {
+            exchange.getExchangeExtension().addOnCompletion(new Synchronization() {
+                @Override
+                public void onComplete(Exchange exchange) {
+                    try {
+                        PahoMqtt5Consumer.this.client.messageArrivedComplete(mqttMessage.getId(), mqttMessage.getQos());
+                    } catch (MqttException e) {
+                        throw new RuntimeException(e);
```

Comment:
dont rethrown but log a WARN about not able to commit

Reply:
Done.